Client/Frontend/Menu/AppMenuConfiguration.swift

```diff
@@ -94,7 +95,10 @@ struct AppMenuConfiguration: MenuConfiguration {
             if #available(iOS 9, *) {
                 menuItems.append(tabState.desktopSite ? AppMenuConfiguration.RequestMobileMenuItem : AppMenuConfiguration.RequestDesktopMenuItem)
             }
-            if HomePageAccessors.hasHomePage(appState) {
+
+            if !HomePageAccessors.isButtonInMenu(appState) {
```

Comment:
I would prefer for the AppState to be the only thing used here to determine menu items. The idea being dependency injection allowing for easy testability of code and mocking etc. I'd rather have a `isChinaRegion` or equivalent flag as the thing that is tested than have another object defined outside of our injected dependencies to query.

I realise that I totally didn't spot that for the other home page stuff, but ditto for `hasHomePage()`